nussl/separation/duet.py

```diff
@@ -113,102 +164,78 @@ def run(self):
+            atn_bins (np.array): attenuation bins
+            delay_bins (np.array): delay bins
+        """
         # calculate the weighted histogram
-        p = 1
-        q = 0
-        tfw = (np.abs(stft_ch0) * np.abs(stft_ch1)) ** p * (np.abs(frequency_matrix)) ** q  # time-freq weights
+        time_frequency_weights = (np.abs(self.stft_ch0) * np.abs(self.stft_ch1)) ** p \
+                                 * (np.abs(self.frequency_matrix)) ** q
 
         # only consider time-freq. points yielding estimates in bounds
-        a_premask = np.logical_and(self.attenuation_min < symmetric_attenuation, symmetric_attenuation < self.attenuation_max)
-        d_premask = np.logical_and(self.delay_min < relative_delay, relative_delay < self.delay_max)
+        a_premask = np.logical_and(self.attenuation_min < self.symmetric_atn, self.symmetric_atn < self.attenuation_max)
+        d_premask = np.logical_and(self.delay_min < self.delay, self.delay < self.delay_max)
         ad_premask = np.logical_and(a_premask, d_premask)
 
         ad_nzind = np.nonzero(ad_premask)
-        alpha_vec = symmetric_attenuation[ad_nzind]
-        delta_vec = relative_delay[ad_nzind]
-        tfw_vec = tfw[ad_nzind]
+        symmetric_atn_vec = self.symmetric_atn[ad_nzind]
+        delay_vec = self.delay[ad_nzind]
+        tfw_vec = time_frequency_weights[ad_nzind]
 
         # compute the histogram
-        H = np.histogram2d(alpha_vec, delta_vec, bins=np.array([self.num_attenuation_bins, self.num_delay_bins]),
-                           range=np.array([[self.attenuation_min, self.attenuation_max],
+        hist, atn_bins, delay_bins = np.histogram2d(symmetric_atn_vec, delay_vec,
+                                                     bins=np.array([self.num_attenuation_bins, self.num_delay_bins]),
+                                                     range=np.array([[self.attenuation_min, self.attenuation_max],
                                            [self.delay_min, self.delay_max]]), normed=False,
                            weights=tfw_vec)
 
-        hist = H[0] / H[0].max()
-        agrid = H[1]
-        dgrid = H[2]
-
         # Save these for later
-        self.attenuation_grid = agrid
-        self.delay_grid = dgrid
-        self.attenuation_delay_histogram = hist
-        self.non_normalized_hist = H[0]
-        self.smoothed_hist = self._smooth_matrix(self.non_normalized_hist, np.array([3]))
+        self.attenuation_grid = atn_bins
+        self.delay_grid = delay_bins
 
-        # smooth the histogram - local average 3-by-3 neighboring bins
-        hist = self._smooth_matrix(hist, np.array([3]))
+        # TODO: do these operations happen twice?
+        # resolved
+        self.attenuation_delay_histogram = hist / hist.max()
+        self.non_normalized_hist = hist
+
+        # smooth the normalized histogram - local average 3-by-3 neighboring bins
+        hist = self._smooth_matrix(self.attenuation_delay_histogram, np.array([3]))
```

Comment:
Did we talk about this? Why is there both `hist` and `self.attenuation_delay_histogram`? **Don't rewrite code!!!** If you've already calculated the thing you need, use it, don't recalculate it!